src/olympia/blocklist/mlbf.py

```diff
@@ -240,20 +240,37 @@ def generate_diffs(
         }
 
     def generate_and_write_stash(self, previous_mlbf: 'MLBF' = None):
-        # compare previous with current blocks
-        extras, deletes, _ = self.generate_diffs(previous_mlbf)[BlockType.BLOCKED]
+        """
+        Generate and write the stash file representing
+        changes between the previous and current MLBF Filters.
+        In order to support older FX clients that only understood blocked and unblocked
+        The unblocked list is a union of deletions from blocked and soft_blocked.
```

Comment:
Where is this coming from?

Reply:
Unblocked includes versions which have gone from hard blocked or from soft blocked to unblocked. It also includes versions which have gone from hard blocked to soft blocked.

Reply:
This is attempting to implement what is written in the google doc you wrote.

Reply:
I updated the doc string to be a bit clearer. lmk